lib/ansible/runner/credential/google_credential.rb

```diff
@@ -0,0 +1,37 @@
+module Ansible
+  class Runner
+    class GoogleCredential < Credential
+      def self.auth_type
+        "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::GoogleCredential"
+      end
+
+      # Modeled off of gce injectors for awx:
+      #
+      #   https://github.com/ansible/awx/blob/1242ee2b/awx/main/models/credential/injectors.py#L18-L42
+      #
+      def env_vars
+        write_credentials_file
+        { "GCE_CREDENTIALS_FILE_PATH" => gce_credentials_file }
```

Comment:
Do we also need `GCE_EMAIL` and `GCE_PROJECT`?

It actually looks like the json file and these env vars are for two different modules.
In the GCE module there is a [note](https://docs.ansible.com/ansible/latest/modules/gce_module.html#notes) that says "JSON credentials strongly preferred". So I would imagine that this is the target for this json file, but that module is deprecated and set to be removed in ansible 2.12.

The alternative module uses different env vars for authenticating and doesn't mention JSON specifically https://docs.ansible.com/ansible/latest/modules/gcp_compute_instance_module.html#notes. This module has env vars for some of the existing parameters, but they don't match the ones used by AWX.

Reply:
So my problem was with this line:

https://github.com/ansible/awx/blob/devel/awx/main/models/credential/injectors.py#L28

Since with how things are set up, we have no way of knowing what the `env` is when executing these methods.  If we want to stay true to `awx`, I think we would have to change how the `lib/ansible/runner/credential/*` classes are being done and pass in the existing `ENV` context when initializing.

That said, seems like `GCE_EMAIL` and `GCE_PROJECT` are more of the default (since I think the absence of a variable can be assumed more often than not) compared to what I have done, so I probably should have gone with that instead of the 'token_uri' value.  I can change to that if you agree.

Reply:
> I think we would have to change how the lib/ansible/runner/credential/* classes are being done and pass in the existing ENV context when initializing.

I don't think this would help as it's likely that `INVENTORY_UPDATE_ID` is an AWX specific detail. I would imagine that this is distinguishing between fetching inventory (as in hosts to run playbooks against) and running a playbook (probably targeting localhost) which uses the module.

> That said, seems like GCE_EMAIL and GCE_PROJECT are more of the default (since I think the absence of a variable can be assumed more often than not)

Yeah that's what I was thinking. For our use case I don't think we would have the variable set.

Outside of all of that, I still can't seem to square any of the env vars with anything I see in the role documentation. This is one of the ones where I think it will be useful to have a playbook that just prints the environment to work out what's being done in the old version of embedded ansible so we can make this match.

Reply:
Yeah, I kinda was thinking with this that we get these in with a basic level of confidence before next Monday, then start testing against live services and make sure things are trued up between what we had and what we now have.

We can then make any fixes as "bugs" as follow ups.

Reply:
Okay, this should be fixed now.